apps/studio/components/interfaces/Auth/Hooks/CreateHookSheet.tsx

```diff
@@ -0,0 +1,425 @@
+        })
+      } else {
+        form.reset({
+          hookType: title || '',
+          enabled: true,
+          selectedType: 'postgres',
+          httpsValues: {
+            url: '',
+            secret: '',
+          },
+          postgresValues: {
+            schema: 'public',
+            functionName: '',
+          },
+        })
+      }
+    }
+  }, [authConfig, title, visible])
+
+  const onSubmit: SubmitHandler<z.infer<typeof FormSchema>> = async (values) => {
+    const definition = HOOKS_DEFINITIONS.find((d) => values.hookType === d.title)
+
+    if (!definition) {
+      return
+    }
+
+    const enabledLabel = definition.enabledKey
+    const uriLabel = definition.uriKey
+    const secretsLabel = definition.secretsKey
+
+    let url = ''
+    if (values.selectedType === 'postgres') {
+      url = `pg-functions://postgres/${values.postgresValues.schema}/${values.postgresValues.functionName}`
+    } else {
+      url = values.httpsValues.url
+    }
+
+    const payload = {
+      [enabledLabel]: values.enabled,
+      [uriLabel]: url,
+      [secretsLabel]: values.selectedType === 'https' ? values.httpsValues.secret : null,
+    }
+
+    updateAuthConfig(
+      { projectRef: projectRef!, config: payload },
+      {
+        onSuccess: () => {
+          toast.success(`Successfully created ${values.hookType}.`)
+          onClose()
+        },
+      }
+    )
+  }
+
+  const values = form.getValues()
+
+  return (
+    <Sheet open={visible} onOpenChange={() => onClose()}>
+      <SheetContent showClose={false} className="flex flex-col gap-0">
+        <SheetHeader className="py-3 flex flex-row justify-between items-center border-b-0">
+          <div className="flex flex-row gap-3 items-center">
+            <SheetClose
+              className={cn(
+                'text-muted hover:text ring-offset-background transition-opacity hover:opacity-100',
+                'focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2',
+                'disabled:pointer-events-none data-[state=open]:bg-secondary',
+                'transition'
+              )}
+            >
+              <X className="h-3 w-3" />
+              <span className="sr-only">Close</span>
+            </SheetClose>
+            <SheetTitle className="truncate">
+              {isCreating ? `Add ${title}` : `Update ${title}`}
+            </SheetTitle>
+          </div>
+        </SheetHeader>
+        <Separator />
+        <Form_Shadcn_ {...form}>
+          <form
+            id={FORM_ID}
+            className="space-y-6 w-full py-8 flex-1"
+            onSubmit={form.handleSubmit(onSubmit)}
+          >
+            <div>
+              <FormField_Shadcn_
+                key="enabled"
+                control={form.control}
+                name="enabled"
+                render={({ field }) => (
+                  <FormItem_Shadcn_ className="px-8">
+                    <FormControl_Shadcn_>
+                      <Toggle
```

Comment:
pleas use `<Switch/>`

think I better get the deprecation notices in.

Reply:
see link above, shows Switch used in a FormItemLayout component

Reply:
https://github.com/supabase/supabase/pull/26817
deprecation notices PR

Reply:
Fixed.